sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -63,7 +64,7 @@ class SparkSession private(
     @transient private val existingSharedState: Option[SharedState])
   extends Serializable with Logging { self =>
 
-  def this(sc: SparkContext) {
+  private[sql] def this(sc: SparkContext) {
```

Comment:
maybe we should change that and have sqlcontext always getting it from sparksession tiself?